DataScience/ProbabilisticParser/data/createTrainingDataProbabilisticParser.py

```diff
@@ -0,0 +1,290 @@
+#!/usr/bin/env python
+"""
+ONS Address Index - Create Training Data for the Probabilistic Parser
+=====================================================================
+
+A simple script to use AddressBase information to create training and houldout
+samples for probabilistic parser.
+
+The script allows to reformat the delivery point table information so
+that it can be used to train a probabilistic parser. The training data
+need to be in XML format and each address token specified separately.
+The holdout data is drawn from the same underlying data source, but it
+is independent from the training sample, that is there are no addresses
+that appear in both samples.
+
+
+Requirements
+------------
+
+:requires: pandas
+:requires: numpy
+
+
+Running
+-------
+
+After all requirements are satisfied, the script can be invoked using CPython interpreter::
+
+    python createTrainingDataProbabilisticParser.py
+
+
+
+Author
+------
+
+:author: Sami Niemi (sami.niemi@valtech.co.uk)
+
+
+Version
+-------
+
+:version: 0.6
+:date: 6-Feb-2017
+"""
+import numbers
+import numpy as np
+import pandas as pd
+
+# set the random seed so that we get the same training and holdout data even if rerun
+seed = np.random.seed(seed=42)
+# turn of pandas chaining warning
+pd.options.mode.chained_assignment = None
+
+
+def _toXML(row):
+    """
+    Convert pandas DataFrame row to string that is valid XML and can be used
+    for training. Assumes that the data frame columns are in the appropriate
+    order.
+
+    :param row: pandas data frame row containing the address information
+
+    :return: a string containing an address in tokenised format
+    :rtype: str
+    """
+    xml = ['\n   <AddressString>']
+
+    for field in row.index:
```

Comment:
I cannot provide the exact answer, but this looks like a code smell: this could be written in a more declarative way, to begin you could filter `None` from things in `row.index` and then map over things that remain there.